cmd/buck/cli/cli.go

```diff
@@ -200,16 +203,30 @@ var linksCmd = &cobra.Command{
 		}
 		links, err := buck.RemoteLinks(ctx, pth)
 		cmd.ErrCheck(err)
-		printLinks(links)
+
+		jsonFormat, err := c.Flags().GetBool("json")
+		cmd.ErrCheck(err)
+		format, err := c.Flags().GetString("format")
+		cmd.ErrCheck(err)
+
+		if jsonFormat == true {
+			format = "json"
+		}
+
+		printLinks(links, format)
 	},
 }
 
-func printLinks(reply local.Links) {
-	cmd.Message("Your bucket links:")
-	cmd.Message("%s Thread link", aurora.White(reply.URL).Bold())
-	cmd.Message("%s IPNS link (propagation can be slow)", aurora.White(reply.IPNS).Bold())
-	if reply.WWW != "" {
-		cmd.Message("%s Bucket website", aurora.White(reply.WWW).Bold())
+func printLinks(reply local.Links, format string) {
```

Comment:
You could make `format` an enum that has a default value which results in the current behavior, like...
```
type Format string

const(
    Default Format = "default"
    JSON = "json"
)
```

Then the func becomes, `func printLinks(reply local.Links, format Format) {`. You can assert the flag value as `Format` like `Format("json")`.

Reply:
Is it possible to overload functions with these mechanisms?

If the enum has a default value. Is it still required to pass the `format` function parameter?

Reply:
Go doesn't allow overloading like that. You'd have to explicity call `printLinks(links, Default)`.

Reply:
It's typically best practice to append the type name to enum values, so the consts would become,
```
type Format string

const(
    DefaultFormat Format = "default"
    JSONFormat = "json"
)
```